### ISSUE TITLE
Add new rule: `remark-lint-consistent-linebreaks`

### ISSUE THREAD

**Issue by remcohaszing** [remarkjs/remark-lint#292]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

Commonmark supports two ways to add a linebreak.

Using two spaces at the end of a line:

```md
foo··
bar
```

Using a backslash at the end of a line:

```md
foo\
bar
```

It would be nice if remark could enforce consistency.

### Solution

Add a new rule: `remark-lint-consistent-linebreaks`

This rule enforces one style or the other, defaulting to `\`.

### Alternatives

People shouldn’t use line breaks in the first place, but since they do, let’s prefer the visible one over the invisible one :shrug:

**Comment by wooorm** [comment]
Good idea!
I think it should be called `remark-lint-hard-break-style`:

- The word “hard break” is more often used to differentiate from a soft line break, it’s also used for the existing rule `remark-lint-hard-break-spaces`
- `linebreak` is used for the existing rule `remark-lint-linebreak-style`, which checks for CRLF/LF
- the word “consistent” is used in this project as an option, which means that the first used style is chosen, and later uses should be the same. This is typically the default.

I think it should have the options:

- `'consistent'` (default)
- `'escape'` (accepting e.g. `a\(?:\r|\r\n|\n)b`)
- `'trailing'` (accepting e.g. `a  (?:\r|\r\n|\n)b`)

**Comment by wooorm** [comment]
Oh this was 3817b19 btw. At some point I want to flip that option in a major, hard breaks can made with escapes are typically recommended!
